clouddrift/adapters/utils.py

```diff
@@ -205,9 +205,9 @@ def _download_with_progress(
 
             if isinstance(temp_output, str) and os.path.getsize(temp_output) > 0:
                 # (temp_output = str <-> output = str)
-                if os.path.exists(output): # type: ignore
-                    os.remove(output) # type: ignore
-                os.rename(temp_output, output) # type: ignore
+                if os.path.exists(output):  # type: ignore
+                    os.remove(output)  # type: ignore
+                os.rename(temp_output, output)  # type: ignore
```

Comment:
Mypy doesn't recognize that because temp_output is a str that output is a str. The definition of temp_out comes from determining is output is a str. I.e. if output is a str, define temp_output as a str.